### ISSUE TITLE
Can support new bge-reranker-v2-m3 ?

### ISSUE THREAD

**Issue by hanswang73** [huggingface/text-embeddings-inference#233]
### Feature request

support new bge-reranker-v2-m3 

### Motivation

support new bge-reranker-v2-m3 

### Your contribution

N

**Comment by OlivierDehaene** [comment]
It is already supported.

**Comment by xunfeng1980** [comment]
## run
```
docker run --rm-p 8080:80  ghcr.io/huggingface/text-embeddings-inference:cpu-1.2.2  --model-id  BAAI/bge-reranker-v2-m3
```
## error
```
24-04-17T04:14:34.383276Z  INFO text_embeddings_router: router/src/main.rs:140: Args { model_id: "BAA*/***-********-*2-m3", revision: None, tokenization_workers: None, dtype: None, pooling: Some(Splade), max_concurrent_requests: 512, max_batch_tokens: 16384, max_batch_requests: None, max_client_batch_size: 32, auto_truncate: false, hf_api_token: None, hostname: "8afa6bb88b28", port: 80, uds_path: "/tmp/text-embeddings-inference-server", huggingface_hub_cache: Some("/data"), payload_limit: 2000000, api_key: None, json_output: false, otlp_endpoint: None, cors_allow_origin: None }
2024-04-17T04:14:34.383331Z  INFO hf_hub: /usr/local/cargo/git/checkouts/hf-hub-1aadb4c6e2cbe1ba/b167f69/src/lib.rs:55: Token file not found "/root/.cache/huggingface/token"    
2024-04-17T04:14:36.120734Z  INFO download_artifacts: text_embeddings_core::download: core/src/download.rs:20: Starting download
2024-04-17T04:14:36.122038Z  INFO download_artifacts: text_embeddings_core::download: core/src/download.rs:37: Model artifacts downloaded in 1.306753ms
2024-04-17T04:14:36.124450Z  WARN text_embeddings_router: router/src/lib.rs:316: `--pooling` arg is set but model is a classifier. Ignoring `--pooling` arg.
2024-04-17T04:14:36.379354Z  WARN text_embeddings_router: router/src/lib.rs:165: Could not find a Sentence Transformers config
2024-04-17T04:14:36.379367Z  INFO text_embeddings_router: router/src/lib.rs:169: Maximum number of tokens per request: 8192
2024-04-17T04:14:36.398158Z  INFO text_embeddings_core::tokenization: core/src/tokenization.rs:23: Starting 10 tokenization workers
2024-04-17T04:14:37.526753Z  INFO text_embeddings_router: router/src/lib.rs:194: Starting model backend
2024-04-17T04:14:37.527274Z  INFO text_embeddings_backend_candle: backends/candle/src/lib.rs:122: Starting Bert model on Cpu
2024-04-17T04:14:38.330844Z  WARN text_embeddings_router: router/src/lib.rs:263: Invalid hostname, defaulting to 0.0.0.0
2024-04-17T04:14:38.331818Z  INFO text_embeddings_router::http::server: router/src/http/server.rs:1553: Starting HTTP server: 0.0.0.0:80
2024-04-17T04:14:38.331826Z  INFO text_embeddings_router::http::server: router/src/http/server.rs:1554: Ready
2024-04-17T04:14:41.059712Z ERROR rerank:predict{inputs=("What is Deep Learning?", "Deep Learning is not...") truncate=false raw_scores=false}: text_embeddings_core::infer: core/src/infer.rs:392: MatMulUnexpectedStriding { lhs_l: Layout { shape: [2, 1024], stride: [14336, 1], start_offset: 0 }, rhs_l: Layout { shape: [1024, 1024], stride: [1, 1024], start_offset: 0 }, bmnk: (1, 2, 1024, 1024), msg: "non-contiguous lhs" }
2024-04-17T04:14:41.059745Z ERROR rerank:predict{inputs=("What is Deep Learning?", "Deep learning is...") truncate=false raw_scores=false}: text_embeddings_core::infer: core/src/infer.rs:392: MatMulUnexpectedStriding { lhs_l: Layout { shape: [2, 1024], stride: [14336, 1], start_offset: 0 }, rhs_l: Layout { shape: [1024, 1024], stride: [1, 1024], start_offset: 0 }, bmnk: (1, 2, 1024, 1024), msg: "non-contiguous lhs" }

```

**Comment by sir3mat** [comment]
Did you solve this issue?

**Comment by xunfeng1980** [comment]
> Did you solve this issue? I solve a similar problem by downgrading to v1.2.0

1.2.1  work on gpu

**Comment by OlivierDehaene** [comment]
1.2.2 work on GPU too.
It is fixed on main, I will do a 1.2.3 release.
